onnxruntime/test/webgpu/external_dawn/main.cc

```diff
@@ -0,0 +1,73 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-FileCopyrightText: Copyright 2024 Arm Limited and/or its affiliates <open-source-office@arm.com>
+// Licensed under the MIT License.
+
+#include <iostream>
+
+#include "core/session/onnxruntime_cxx_api.h"
+
+#include <google/protobuf/stubs/common.h>
+
+#include "dawn/native/DawnNative.h"
+
+#ifdef _WIN32
+int wmain(int argc, wchar_t* argv[]) {
+#else
+int main(int argc, char* argv[]) {
+#endif
+  bool no_proc_table = argc > 0 &&
+#ifdef _WIN32
+                       wcscmp(L"--no_proc_table", argv[argc - 1]) == 0;
+#else
+                       strcmp("--no_proc_table", argv[argc - 1]) == 0;
+#endif
+
+  int retval = 0;
+  Ort::Env env{nullptr};
+  try {
+    env = Ort::Env{ORT_LOGGING_LEVEL_WARNING, "Default"};
+
+    // model is https://github.com/onnx/onnx/blob/v1.15.0/onnx/backend/test/data/node/test_abs/model.onnx
+    constexpr uint8_t MODEL_DATA[] = {8, 7, 18, 12, 98, 97, 99, 107, 101, 110,
```

Comment:
A tiny suggestion:  std::array is prefered than raw Carrays. 

constexpr std::array<uint8_t, length> MODEL_DATA{8,7,... };

std::array's interface is very like std::vector. It has .data() and .size() functions.  And in debug build mode it has range check which C array doesn't have.

Reply:
The only thing that I don't like std::array is before C++20 the length need to be specified explicitly.

Reply:
Yeah, it is slightly inconvenient, but I think overall std::array is better than raw arrays.  Just a suggestion. You don't have to change this code.